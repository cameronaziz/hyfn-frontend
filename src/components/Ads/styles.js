const styles = theme => ({
  root: {
    width: '80%',
    margin: '0 auto',
    marginTop: theme.spacing.unit * 3,
    overflowX: 'auto',
  },
  stickyCol: {
    left: 0,
    position: 'sticky',
    backgroundColor: '#fff',
  },
  table: {
    minWidth: 500,
  },
  tableHead: {
    whiteSpace: 'nowrap',
  },
  tableRow: {

  },
});

export default styles;
