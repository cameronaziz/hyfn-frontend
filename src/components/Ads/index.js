import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import {
  Paper,
  Table,
  TableBody,
  TableCell,
  TableHead,
  TableRow,
  Chip,
} from '@material-ui/core';
import { graphql } from 'react-apollo';
import { getAds } from '../../queries';
import styles from './styles';

const Ads = ({
  adSelected, classes, selectAd, data,
}) => {
  if (data.loading) { return <div>Loading</div>; }
  return (
    <Fragment>
      <h1>Data</h1>
      <Paper className={classes.root}>
        <Table className={classes.table}>
          <TableHead className={classes.tableHead}>
            <TableRow>
              <TableCell className={classes.stickyCol}>Name</TableCell>
              <TableCell className={classes.col}>Remote ID</TableCell>
              <TableCell>Status</TableCell>
              <TableCell>Frequency</TableCell>
              <TableCell>Spend</TableCell>
              <TableCell>CPM</TableCell>
              <TableCell>CTR</TableCell>
              <TableCell>Cost per Inline Link Click</TableCell>
              <TableCell>Goal</TableCell>
              <TableCell>Link Click</TableCell>
              <TableCell>Cost per Goal</TableCell>
              <TableCell>Offsite Conversion</TableCell>
              <TableCell>Status</TableCell>
              <TableCell>Reach</TableCell>
            </TableRow>
          </TableHead>
          <TableBody>
            {data.adsDetails && data.adsDetails.map(row => (
              <TableRow
                key={row.id}
                className="ad-row"
                selected={adSelected === row.id}
                onClick={() => {
                  selectAd(row.id);
                }}
              >
                <TableCell className={classes.stickyCol}>
                  {row.name}
                </TableCell>
                <TableCell>{row.remoteId}</TableCell>
                <TableCell>
                  <Chip
                    color={
                      row.status === 'ACTIVE' ? 'primary' : 'secondary'
                    }
                    label={row.status}
                  />
                </TableCell>
                <TableCell>{row.frequency}</TableCell>
                <TableCell>{row.spend}</TableCell>
                <TableCell>{row.cpm}</TableCell>
                <TableCell>{row.ctr}</TableCell>
                <TableCell>{row.costPerInlineLinkClick}</TableCell>
                <TableCell>{row.goal}</TableCell>
                <TableCell>{row.linkClick}</TableCell>
                <TableCell>{row.costPerGoal}</TableCell>
                <TableCell>{row.offsiteConversion}</TableCell>
                <TableCell>{row.reach}</TableCell>
              </TableRow>
            ))}
          </TableBody>
        </Table>
      </Paper>
    </Fragment>
  );
};

Ads.propTypes = {
  selectAd: PropTypes.func.isRequired,
  classes: PropTypes.shape({
    root: PropTypes.string.isRequired,
    table: PropTypes.string.isRequired,
    tableHead: PropTypes.string.isRequired,
    stickyCol: PropTypes.string.isRequired,
    tableRow: PropTypes.string.isRequired,
  }).isRequired,
  adSelected: PropTypes.string,
};

Ads.defaultProps = {
  adSelected: undefined,
};

export default graphql(getAds)(withStyles(styles)(Ads));
