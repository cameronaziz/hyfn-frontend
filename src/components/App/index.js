import React, { Component } from 'react';
import logo from './logo.svg';
import Ads from '../Ads';
import './App.css';

class App extends Component {
  state = { adSelected: undefined }

  selectAd = (adId) => {
    const adSelected = adId.toString();
    this.setState({
      adSelected,
    });
  }

  render() {
    const { adSelected } = this.state;
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
        </header>
        <Ads selectAd={this.selectAd} adSelected={adSelected} />
      </div>
    );
  }
}

export default App;
