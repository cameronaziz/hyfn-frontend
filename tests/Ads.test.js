import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { MockedProvider } from 'react-apollo/test-utils';
import { ApolloProvider } from 'react-apollo';
import { client } from './apollo';
import * as mocks from './mocks';
import Ads from '../src/components/Ads';
import App from '../src/components/App';
import './adapter';

describe('Ads Component', () => {
  const appComponent = (
    <MockedProvider addTypename={false} mocks={[mocks.getAds]}>
      <App />
    </MockedProvider>
  );

  const appWrapper = mount(appComponent);
  const app = appWrapper.find(App).instance();
  const selectAd = jest.spyOn(app, 'selectAd');

  const adsComponent = (
    <MockedProvider addTypename={false} mocks={[mocks.getAds]}>
      <Ads selectAd={selectAd} />
    </MockedProvider>
  );

  it('will match snapshot', () => {
    const renderedComponent = renderer.create(
      <ApolloProvider client={client}>
        <Ads adSelected={undefined} selectAd={selectAd} />
      </ApolloProvider>,
    );
    const tree = renderedComponent.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('will load API', () => {
    const wrapper = mount(adsComponent);
    expect(wrapper.contains(<div>Loading</div>)).toEqual(true);
  });

  it('will render data', async () => {
    const wrapper = mount(adsComponent);
    expect(wrapper.find('.ad-row').exists()).toEqual(false);
    await new Promise(resolve => setTimeout(resolve));
    wrapper.update();
    expect(wrapper.find('.ad-row').exists()).toEqual(true);
  });

  it('will click and execute adSelected', async () => {
    const adsWrapper = mount(adsComponent);
    await new Promise(resolve => setTimeout(resolve));
    adsWrapper.update();
    const firstElement = adsWrapper.find('.ad-row').first();
    firstElement.first().simulate('click');
    await new Promise(resolve => setTimeout(resolve));
    appWrapper.update();
    expect(selectAd).toHaveBeenCalled();
  });
});
