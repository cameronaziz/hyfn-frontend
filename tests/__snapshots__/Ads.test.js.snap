// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ads Component will match snapshot 1`] = `
<div>
  Loading
</div>
`;
